src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy

```diff
@@ -24,16 +24,33 @@ class VersionResolver {
     private final ScmRepository repository
 
     private final VersionSorter sorter
+    private ScmPosition latestChangePosition
+    private String projectRootRelativePath
+    private List<String> foldersToExcludeWhenResolvingLatestRelevantRootProjectCommit
 
     VersionResolver(ScmRepository repository) {
+        this(repository, null, Collections.emptyList())
+    }
+
+    VersionResolver(ScmRepository repository, String projectRootRelativePath) {
+        this(repository, projectRootRelativePath, Collections.emptyList())
+    }
+
+    VersionResolver(ScmRepository repository, String projectRootRelativePath, List<String> excludeFolders) {
         this.repository = repository
+        this.projectRootRelativePath = projectRootRelativePath
         this.sorter = new VersionSorter()
+        this.foldersToExcludeWhenResolvingLatestRelevantRootProjectCommit = excludeFolders
     }
 
     VersionContext resolveVersion(VersionProperties versionRules, TagProperties tagProperties, NextVersionProperties nextVersionProperties) {
-        ScmPosition position = repository.currentPosition()
+        if (projectRootRelativePath != null) {
```

Comment:
from what i see `projectRootRelativePath` is never `null` on production (might be null in tests) - why keep this switch? it makes prod code more complicated and i don't want tests to complicate prod code

Reply:
also: this is the place where old vs new flow can be distinguished using feature switch i proposed :)

Reply:
Yes the check was to see if the path was `null` as a result of tests, which it no longer can be due to the changes of constructor.  However, all resolution of versions must go through the new method, as the project may be running as a submodule, i.e. `:submodule:currentVersion`, as opposed to `:currentversion`.  The configuration parsed by that submodule is its own config and not that of the parent project, which is where the feature flag to enable monorepo support would be set (i.e. the subprojects would also need to have the feature flag set to true in their own config). I don't think it's correct that submodules should have to also have the flag set. This means that there's no need for a feature flag.